packages/wrapped-keys/src/lib/api/store-encrypted-key-batch.ts

```diff
@@ -0,0 +1,54 @@
+import { storePrivateKeyBatch } from '../service-client';
+import { StoreKeyBatchParams } from '../service-client/types';
+import {
+  StoredKeyData,
+  StoreEncryptedKeyBatchParams,
+  StoreEncryptedKeyBatchResult,
+} from '../types';
+import { getFirstSessionSig, getPkpAddressFromSessionSig } from '../utils';
+
+/** Stores a batch of encrypted private keys and their metadata to the wrapped keys backend service
+ *
+ * @param { StoreEncryptedKeyParams } params Parameters required to fetch the encrypted private key metadata
+ * @returns { Promise<StoreEncryptedKeyResult> } The encrypted private key and its associated metadata
+ */
+export async function storeEncryptedKeyBatch(
+  params: StoreEncryptedKeyBatchParams
+): Promise<StoreEncryptedKeyBatchResult> {
+  const { pkpSessionSigs, litNodeClient, keyBatch } = params;
+  const sessionSig = getFirstSessionSig(pkpSessionSigs);
+  const pkpAddress = getPkpAddressFromSessionSig(sessionSig);
+
+  const storedKeyMetadataBatch: StoreKeyBatchParams['storedKeyMetadataBatch'] =
+    keyBatch.map(
+      ({
+        keyType,
+        publicKey,
+        memo,
+        dataToEncryptHash,
+        ciphertext,
+      }): Pick<
+        StoredKeyData,
+        | 'pkpAddress'
+        | 'publicKey'
+        | 'keyType'
+        | 'dataToEncryptHash'
+        | 'ciphertext'
+        | 'memo'
+      > => ({
+        pkpAddress,
+        publicKey,
+        memo,
+        dataToEncryptHash,
+        ciphertext,
+        keyType,
+      })
+    );
+
+  return storePrivateKeyBatch({
+    pkpAddress,
```

Comment:
Why is this required isn't the `pkpAddress` extracted from the PKP sessionSig?

Reply:
Good question! I checked into this, and it was a bit of a rabbit hole -- I realised that this method _doesn't_ need it -- but `list` and `get` do -- I _think_ it just ended up in all of the `service-client` methods dependencies parasitically even though we don't always need it.

This question unearthed an issue I hadn't noticed before, however -- we were importing the `getPkpAddressFromSessionSig()` method into `listPrivateKeyMetadata()` and `fetchPrivateKey()` _but we shouldn't have been_  the original design intention for `wrapped-keys` was that these operations would always be handled in the `api` layer and the result of them would be passed into the `service-client`, so we never ran those methods multiple times unnecessarily while servicing a `api` client call.

Thankfully, everything in the `service-client` is private so I fixed this problem and removed the `pkpAddress` parameter from the `service-client` parameter types that it was not needed for.

Fixing this encapsulation issue also exposed that we had imported the `service-client` and the internal util method into our `testImportPrivateKey` local-test -- so I fixed that too ;)

Good catch!